### PR TITLE
[docs] Add documentation on how to handle read-only arrays and actor reprs

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

This is a common issue that has come up a few times in the forums, and we don't have an explicit example.

### CODE REVIEW

File: doc/source/ray-core/doc_code/deser.py
Comment (zhe-thoughts) [inline doc/source/ray-core/doc_code/deser.py:3]: Nit (and not blocking this PR): We are not consistent on whether to add `ray.init()` in code snippets in docs.

File: doc/source/ray-core/objects/serialization.rst
Comment (zhe-thoughts) [inline doc/source/ray-core/objects/serialization.rst:35]: (Not blocking this PR): is it possible to address the issue on API level?

File: doc/source/ray-observability/ray-logging.rst
Comment (zhe-thoughts) [inline doc/source/ray-observability/ray-logging.rst:48]: Nit: add a link to https://docs.python.org/3/library/functions.html#repr? So that people understand they are overriding a Python built-in method.

Question (not blocking the PR): is this scenario common enough such that we should create something like a superclass called `IndexedActor`?

**Review by zhe-thoughts — APPROVED** [review]
LGTM (pending a nit on `repr` link).

File: doc/source/ray-core/doc_code/deser.py
Comment (ericl) [reply doc/source/ray-core/doc_code/deser.py:3]: I think we should avoid, since it's done implicitly now and makes the code longer.

File: doc/source/ray-core/objects/serialization.rst
Comment (ericl) [reply doc/source/ray-core/objects/serialization.rst:35]: We would have to think about how to do it, but possible. Maybe something like `.options(copy_args=True)` or similar.

File: doc/source/ray-observability/ray-logging.rst
Comment (ericl) [reply doc/source/ray-observability/ray-logging.rst]: Done. Hmm, I think the use cases have been one-off enough an indexed actor would generally need to be customized anyway.

Comment (zhe-thoughts) [reply doc/source/ray-observability/ray-logging.rst]: Sounds good. Thanks

**Pull request merged**
